### ISSUE TITLE
Incorrect units for memory comparison in docker test-client

### ISSUE THREAD

**Issue by avgojzack** [strimzi/strimzi-kafka-operator#4223]
The function that gets the heap size for the operators has an incorrect value for exponent using in "DEFAULT_MEMORY_CEILING". There are 20**30 bytes in a gigabyte. The result is that a value of this function is never returned and the heap size always defaults to 1gb for the operators that use this function.

https://github.com/strimzi/strimzi-kafka-operator/blob/master/docker-images/test-client/scripts/dynamic_resources.sh

Please note, this problem is only in the "test-client", so I don't know how serious it is.

```
function get_heap_size {
  CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
   # use max of 31G memory, java performs much better with Compressed Ordinary Object Pointers
  DEFAULT_MEMORY_CEILING=$((31 * 2**20))
  if [ "${CONTAINER_MEMORY_IN_BYTES}" -lt "${DEFAULT_MEMORY_CEILING}" ]; then
    if [ -z $CONTAINER_HEAP_PERCENT ]; then
      CONTAINER_HEAP_PERCENT=0.50
    fi

    CONTAINER_MEMORY_IN_MB=$((${CONTAINER_MEMORY_IN_BYTES}/1024**2))
    CONTAINER_HEAP_MAX=$(echo "${CONTAINER_MEMORY_IN_MB} ${CONTAINER_HEAP_PERCENT}" | awk '{ printf "%d", $1 * $2 }')

    echo "${CONTAINER_HEAP_MAX}"
  fi
}
```

**To Reproduce**
(Sorry, I don't understand how the test-client relates to normal operations, I've just observed it while debugging)

**Environment (please complete the following information):**
 - Strimzi version: master
 - Installation method: Helm chart (copied from yaml files)
 - Kubernetes cluster: 1.17
 - Infrastructure: GCP

**Additional context**
Imagine you have a 1gb limit set:

1gb =1,073,741,824
$((31 * 2**20)) = 32,505,856

1,073,741,824 is not less than 32,505,856, so no value is returned from the function which leaves java defaulting the heap size. No matter how much you increase the limit, you always get the default jvm settings.

**Comment by scholzj** [comment]
You are right. Thanks for raising this. AFAIK this is not used by any operators, just by some test clients. I do not think you should see it anywhere in debugging outside of system tests.

**Comment by scholzj** [comment]
@Frawless This script was removed from operators in #2689 - is there actually any reason for it in the test clients? If not, maybe it can be just removed as well?

**Comment by avgojzack** [comment]
I think we just figured that out, we're on version .17 which looks like it has a separate dynamic_resources.sh for the operators:
https://github.com/strimzi/strimzi-kafka-operator/blob/release-0.17.x/docker-images/operator/scripts/dynamic_resources.sh

**Comment by scholzj** [comment]
Yeah, that makes sense since we removed it in 0.18.

**Comment by scholzj** [comment]
Fixed in #4245. Thanks for raising this.
